index.bs

```diff
@@ -345,10 +342,13 @@ When this method is invoked, the user agent MUST execute the following algorithm
 2. Let |promise| be a new <a data-lt="Promises">Promise</a>. Return |promise| and start a timer for |adjustedTimeout| seconds.
     Then asynchronously continue executing the following steps.
 
-3. Set |callerOrigin| to the <a link-for='web'>origin</a> of the caller. Derive the RP ID from |callerOrigin| by computing the
-    "public suffix + 1" or "PS+1" (which is also referred to as the "Effective Top-Level Domain plus One" or "<a>eTLD+1</a>")
-    part of |callerOrigin| [[PSL]]. Let |rpId| be the lowercase form of this RP ID. Set |rpIdHash| to the SHA-256 hash of the
-    UTF-8 encoding of |rpId|.
+3. Set |callerOrigin| to the <a link-for='web'>origin</a> of the caller. If {{CredentialOptions/rpId}} is not specified, then
+    set |rpId| to the Unicode serialization of |callerOrigin| as specified in [[!HTML51-20160621]] section 6.4. If
+    {{CredentialOptions/rpId}} is specified, then check if its value would be an acceptable setting for the current document's
+    `domain` by running the algorithm in [[!HTML51-20160621]] section 6.4.1 (but do not change the current document's `domain`).
+    If it is not permissible, reject |promise| with a <a>DOMException</a> whose name is "SecurityError", and terminate this
+    algorithm. If it is permissible, then set |rpId| to the Unicode serialization of {{CredentialOptions/rpId}}. Set |rpIdHash|
+    to the SHA-256 hash of the UTF-8 encoding of |rpId|.
```

Comment:
I think we want to write the above set of lines more like so:
>If {{CredentialOptions/rpId}} is specified, then invoke the <a link-for='web'>relaxing the same-origin restriction</a> "setting" algorithm using {{CredentialOptions/rpId}} as the `given value` without setting <a link-for='web'>origin</a>'s |domain|. If no errors are thrown, then set |rpId| to the value of |host| as yielded by the algorithm. Set |rpIdHash| to the SHA-256 hash of |rpId|. 

given that the internal default representation of `origin.host` and `origin.domain` is `asciiDomain` ( U-labels have been converted to A-labels (see https://tools.ietf.org/html/rfc5890#section-2.3.2 and foregoing figure 1 on pg 10 ) ) format, we should not be stipulating UTF-8 encoding of them, AFAICT. 

The jeffh-origin-et-al branch will soon have the `<pre class="anchors">` anchors set up to properly ref these things in HTML51 and so once both branches are eventually merged we should be good.